### PR TITLE
Stricter lint rules

### PR DESCRIPTION
**Pull request by avishayt**

### CODE REVIEW

File: cmd/devicesimulator/main.go [hunk 1]
```diff
@@ -96,7 +96,7 @@ func main() {
 			defer wg.Done()
 
 			// stagger the start of each agent
-			time.Sleep(time.Duration(rand.Float64() * float64(*statusUpdateInterval)))
+			time.Sleep(time.Duration(rand.Float64() * float64(*statusUpdateInterval))) //nolint:gosec
```
Comment (hexfusion) [inline cmd/devicesimulator/main.go:99]: can we add a TODO to replace math/rand with crypto/rand?

File: internal/agent/agent.go [hunk 2]
```diff
@@ -375,6 +375,7 @@ func (a *DeviceAgent) ensureManagementClient() error {
 				TLSClientConfig: &tls.Config{
 					RootCAs:      a.caCertPool,
 					Certificates: []tls.Certificate{*a.managementClientCert},
+					MinVersion:   tls.VersionTLS12,
```
Comment (fzdarsky) [inline internal/agent/agent.go:378]: The server uses min TLS 1.3: https://github.com/flightctl/flightctl/blob/49a37ef878677399c7aa6ff7475bddf08156c968/internal/crypto/tls.go#L32

File: cmd/devicesimulator/prometheus.go [hunk 3]
```diff
@@ -54,7 +55,7 @@ var (
 
 func setupMetricsEndpoint(metricsAddress string) {
 	http.Handle("/metrics", promhttp.Handler())
-	srv := &http.Server{Addr: metricsAddress}
+	srv := &http.Server{Addr: metricsAddress, ReadHeaderTimeout: time.Second}
```
Comment (hexfusion) [inline cmd/devicesimulator/prometheus.go:58]: nit: Can we move the timeout to a const?

File: internal/agent/enrollment.go [hunk 4]
```diff
@@ -136,6 +136,7 @@ func (a *DeviceAgent) ensureEnrollmentClient() error {
 				TLSClientConfig: &tls.Config{
 					RootCAs:      a.caCertPool,
 					Certificates: []tls.Certificate{*a.enrollmentClientCert},
+					MinVersion:   tls.VersionTLS12,
```
Comment (fzdarsky) [inline internal/agent/enrollment.go:139]: The server uses a min TLS1.3: https://github.com/flightctl/flightctl/blob/49a37ef878677399c7aa6ff7475bddf08156c968/internal/crypto/tls.go#L32

File: cmd/flightctl/main.go [hunk 5]
```diff
@@ -238,7 +237,7 @@ func getClient() (*client.ClientWithResponses, error) {
 	return client.NewClientWithResponses(serverUrl, client.WithHTTPClient(httpClient))
 }
 
-func RunGet(kind, name string, labelSelector, fleetName *string, output string, limit *int32, cont *string) error {
+func RunGet(kind, name string, labelSelector, fleetName *string, output string, limit *int32, cont *string) error { //nolint:gocyclo
```
Comment (hexfusion) [inline cmd/flightctl/main.go:240]: gocyclo should probably all be fixed  vs nolint?

**Review by hexfusion — COMMENTED** [review]
few notes, great addition.

File: internal/crypto/tls.go [hunk 6]
```diff
@@ -54,5 +54,6 @@ func TLSConfigForClient(caConfig, clientConfig *TLSCertificateConfig) (*tls.Conf
 	return &tls.Config{
 		RootCAs:      caPool,
 		Certificates: []tls.Certificate{cert},
+		MinVersion:   tls.VersionTLS12,
```
Comment (fzdarsky) [inline internal/crypto/tls.go:57]: The server uses min TLS1.3: https://github.com/flightctl/flightctl/blob/49a37ef878677399c7aa6ff7475bddf08156c968/internal/crypto/tls.go#L32

File: cmd/flightctl/main.go (hunk 5 above)
Comment (avishayt) [reply cmd/flightctl/main.go:240]: It's a big refactor, I'll open a separate ticket

Comment (hexfusion) [reply cmd/flightctl/main.go:240]: I guess an issue to track would work if the work is complex.

Comment (avishayt) [reply cmd/flightctl/main.go:240]: Opened [MGMT-16746](https://issues.redhat.com/browse/MGMT-16746)

File: internal/agent/agent.go (hunk 2 above)
Comment (avishayt) [reply internal/agent/agent.go:378]: Fixed

File: internal/crypto/tls.go (hunk 6 above)
Comment (avishayt) [reply internal/crypto/tls.go:57]: Fixed

**Pull request merged**